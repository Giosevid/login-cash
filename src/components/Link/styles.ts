import {StyleSheet, TextStyle, ViewStyle} from 'react-native';
import colors from '../../assets/colors';

type StyleProps = {
  text: TextStyle;
  container: ViewStyle;
};

const styles = StyleSheet.create<StyleProps>({
  container: {
    marginBottom: 16,
  },
  text: {
    color: colors.primary,
    marginTop: 15,
  },
});

export default styles;
