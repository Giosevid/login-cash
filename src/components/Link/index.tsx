import {TouchableOpacity, Text} from 'react-native';
import React from 'react';
import styles from './styles';

type Props = {
  label: string;
  onPress: () => void;
};

const Link = ({label, onPress}: Props) => {
  return (
    <TouchableOpacity onPress={onPress} style={styles.container}>
      <Text style={styles.text}>{label}</Text>
    </TouchableOpacity>
  );
};

export default Link;
