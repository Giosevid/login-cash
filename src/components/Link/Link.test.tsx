import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import Link from './index';

describe('Test Link', () => {
  const handleClick = jest.fn();
  test('Snapshot Link', () => {
    const {toJSON} = render(
      <Link onPress={handleClick} label="Test label link" />,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  test('Expect call handleClick', () => {
    const {getByText} = render(
      <Link onPress={handleClick} label="Test label link" />,
    );
    const element = getByText('Test label link');
    fireEvent.press(element);
    expect(handleClick).toBeCalledTimes(1);
  });
});
