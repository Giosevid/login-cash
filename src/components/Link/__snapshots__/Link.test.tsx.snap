// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Link Snapshot Link 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginBottom": 16,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#4d83dd",
        "marginTop": 15,
      }
    }
  >
    Test label link
  </Text>
</View>
`;
